src/telemetry/registerListeners.ts

```diff
@@ -5,25 +5,32 @@
 
 import { TextDocument, workspace } from 'vscode';
 import { IActionContext, registerEvent } from 'vscode-azureextensionui';
-import { ext } from './extensionVariables';
-
-let lastUploadTime: number = 0;
-const hourInMilliseconds = 1000 * 60 * 60;
+import { ext } from '../extensionVariables';
 
 export function registerListeners(): void {
     if (ext.telemetryOptIn) {
         registerEvent('dockerfilesave', workspace.onDidSaveTextDocument, async (context: IActionContext, doc: TextDocument) => {
-            // If it's not a Dockerfile, or last upload time is within an hour, skip
-            if (doc.languageId !== 'dockerfile' || lastUploadTime + hourInMilliseconds > Date.now()) {
+            // If it's not a Dockerfile, skip
+            if (doc.languageId !== 'dockerfile') {
+                context.telemetry.suppressAll = true;
+                return;
+            }
+
+            context.telemetry.properties.lineCount = doc.lineCount.toString();
+
+            void ext.activityMeasurementService.recordActivity('overall');
+        });
+
+        registerEvent('composefilesave', workspace.onDidSaveTextDocument, async (context: IActionContext, doc: TextDocument) => {
+            // If it's not a compose file, skip
+            if (!/docker-compose.*\.(yml|yaml)/i.test(doc.fileName)) {

```

Comment:
This will soon need to change to include `compose.yaml` too. Not sure if you want to change it now, put a `TODO`, or something else.

Reply:
Yeah. I wrote this code before picking up that other item. :smile: